### DISCUSSION TITLE
Click delay with custom button

### DISCUSSION THREAD

**Discussion by FloralSoda** [slint-ui/slint#6374] [Q&A] [answered]
Hi guys, I'm new to Slint, and I'm currently learning how to do animations. [I cooked up a 3D button on SlintPad](https://releases.slint.dev/1.8.0/editor/?snippet=export+component+PrettyButton+inherits+Rectangle+%7B%0A++++in-out+property+text+%3C%3D%3E+txt.text%3B%0A++++in-out+property+%3Ccolor%3E+primary%3A+%23555%3B%0A++++in-out+property+%3Ccolor%3E+text_color%3A+%23eee%3B%0A++++in-out+property+%3Ccolor%3E+accent%3A+%230ee%3B%0A++++callback+clicked+%3C%3D%3E+touch.clicked%3B%0A++++min-width%3A+txt.preferred-width+%2B+20px%3B%0A++++height%3A+txt.preferred-height+*+1.8%3B%0A++++clip%3A+false%3B%0A++++border-radius%3A+8px%3B%0A++++pure+function+touch_offset%28%29+-%3E+length+%7B%0A++++++++return+%28touch.pressed+%3F+3px+%3A+touch.has-hover+%3F+1px+%3A+0px%29%3B%0A++++%7D%0A++++Rectangle+%7B%0A++++++++border-radius%3A+root.border-radius%3B%0A++++++++background%3A+touch.pressed+%3F+accent+%3A+touch.has-hover+%3F+accent.darker%2825%25%29+%3A+primary.darker%2825%25%29%3B%0A++++++++height%3A+%28plate.height+%2B+3px%29%3B%0A++++++++y%3A+3.5px%3B%0A++++++++animate+background+%7B%0A++++++++++++duration%3A+30ms%3B%0A++++++++++++easing%3A+ease-out%3B%0A++++++++%7D%0A++++%7D%0A%0A++++plate+%3A%3D+Rectangle+%7B%0A++++++++height%3A+root.height+*+0.72%3B%0A++++++++border-radius%3A+root.border-radius%3B%0A++++++++y%3A+touch_offset%28%29%3B%0A++++++++background%3A+primary%3B%0A++++++++animate+y+%7B%0A++++++++++++duration%3A+30ms%3B%0A++++++++++++easing%3A+ease-out-circ%3B%0A++++++++%7D%0A++++++++border-width%3A+1px%3B%0A++++++++border-color%3A+primary.darker%2810%25%29%3B%0A++++%7D%0A%0A++++txt+%3A%3D+Text+%7B%0A++++++++y%3A+plate.y%3B%0A++++++++color%3A+text_color%3B%0A++++++++vertical-alignment%3A+center%3B%0A++++++++height%3A+plate.height%3B%0A++++%7D%0A%0A++++touch+%3A%3D+TouchArea+%7B+%7D%0A%7D%0A%0A%0Aexport+component+Test+inherits+Window+%7B%0A++++VerticalLayout+%7B%0A++++++++alignment%3A+start%3B%0A++++++++PrettyButton+%7B+text%3A+%22Button%22%3B+%7D%0A++++%7D%0A%7D%0A&style=native), and it seemed to work exactly as I wanted it until I put it into an actual app. No code behind it, just a button inside a vertical layout to make sure everything is working properly. 
Inside the app, the button still worked as intended, but with a very slight delay. Give or take like 10ms. I tried removing the animation, but the delay is still there in the app. This delay appears to be purely visual, though, as attaching code to it makes the effect feel responsive as expected.
My initial instinct is the method I'm using to switch states could be introducing some delay, but I'm copying the method used in the [CustomWidgets part of the docs](https://releases.slint.dev/1.8.0/docs/slint/src/recipes/recipes#custom-widgets) and theirs seems more responsive even despite marginally more complicated maths than me, so I don't think it's that.

Could anyone share thoughts or insight on what might be causing this? My platform is Fedora KDE Wayland. I don't experience this with anything else, whether it be other apps or even other components made in Slint.

**Comment by ogoffart** [comment, marked as answer]
Could it be that the Button is inside a Flickable or some other things that tries to intercept the swipe gesture?
Is it only hapenning with your button, or is it also hapenning if you use a standard button?

**Comment by FloralSoda** [comment]
Turns out that I put it inside a scroll view, as part of my base component for pages. Taking it out of the scroll view does fix the issue.
This animation issue does happen for a button from std-widgets. Is there a property I need to set or some workaround?

**Reply by ogoffart** [reply]
A flickable that can be panned with the touch is adding a 100ms delay.
You can change the `interactive` property of the Flickable

**Reply by FloralSoda** [reply]
100ms seems like a long time to be delaying inputs in the context of animations, but this sounds like a solution I can work with. Thank you for your input and time :)
